ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs

```diff
@@ -76,15 +84,47 @@ public double Size
             }
         }
 
-        private void InitializeProfileImage()
+        /// <summary>
+        /// Gets or sets the avatar image.
+        /// </summary>
+        /// <returns>Image source.</returns>
+        public ImageSource GetProfileImage()
         {
             if (Manager.Account.LoginState == FxA.LoginState.LoggedIn)
             {
                 if (Manager.Account.Config.FxALogin.User.Avatar != null)
                 {
-                    Url = Manager.Account.Config.FxALogin.User.Avatar;
+                    var image = Manager.Cache.Get("avatarImage");
+
+                    if (image != null)
+                    {
+                        return (BitmapImage)image;
+                    }
+                    else
+                    {
+                        CacheItemPolicy policy = new CacheItemPolicy();
+
+                        HttpWebRequest req = (HttpWebRequest)WebRequest.Create(Manager.Account.Config.FxALogin.User.Avatar);
+
+                        try
+                        {
+                            using (HttpWebResponse response = (HttpWebResponse)req.GetResponse())
+                            {
+                                image = new BitmapImage(new Uri(Manager.Account.Config.FxALogin.User.Avatar));
+                            }
+
+                            Manager.Cache.Set("avatarImage", image, policy);
+                            return (BitmapImage)image;
+                        }
+                        catch (Exception e)
+                        {
+                            ErrorHandling.ErrorHandler.Handle(e, ErrorHandling.LogLevel.Debug);
+                        }
+                    }
                 }
             }
+
+            return new BitmapImage(new Uri(System.IO.Path.Combine(Environment.CurrentDirectory, @"..\..\..\src\UI\Resources\Icons\Generic\default-avatar.png")));
```

Comment:
I think these lines of code are kind of similar to those in Manager.cs file. Could you create a method to get the image through avatar url? This method can be used in many places and can be easily to maintain.